packages/medusa/src/strategies/product/import.ts

```diff
@@ -0,0 +1,772 @@
+
+/**
+ * Schema definition for the CSV parser.
+ */
+const CSVSchema: ProductImportCsvSchema = {
+  columns: [
+    // PRODUCT
+    {
+      name: "Product id",
+      mapTo: "product.id",
+    },
+    {
+      name: "Product Handle",
+      mapTo: "product.handle",
+      required: true,
+    },
+    { name: "Product Title", mapTo: "product.title" },
+    { name: "Product Subtitle", mapTo: "product.subtitle" },
+    { name: "Product Description", mapTo: "product.description" },
+    { name: "Product Status", mapTo: "product.status" },
+    { name: "Product Thumbnail", mapTo: "product.thumbnail" },
+    { name: "Product Weight", mapTo: "product.weight" },
+    { name: "Product Length", mapTo: "product.length" },
+    { name: "Product Width", mapTo: "product.width" },
+    { name: "Product Height", mapTo: "product.height" },
+    { name: "Product HS Code", mapTo: "product.hs_code" },
+    { name: "Product Origin Country", mapTo: "product.origin_country" },
+    { name: "Product Mid Code", mapTo: "product.mid_code" },
+    { name: "Product Material", mapTo: "product.material" },
+    // PRODUCT-COLLECTION
+    { name: "Product Collection Title", mapTo: "product.collection.title" },
+    { name: "Product Collection Handle", mapTo: "product.collection.handle" },
+    // PRODUCT-TYPE
+    { name: "Product Type", mapTo: "product.type.value" },
+    // PRODUCT-TAGS
+    {
+      name: "Product Tags",
+      mapTo: "product.tags",
+      transform: (value: string) =>
+        `${value}`.split(",").map((v) => ({ value: v })),
+    },
+    //
+    { name: "Product Discountable", mapTo: "product.discountable" },
+    { name: "Product External ID", mapTo: "product.external_id" },
+    // PRODUCT-SHIPPING_PROFILE
+    { name: "Product Profile Name", mapTo: "product.profile.name" },
+    { name: "Product Profile Type", mapTo: "product.profile.type" },
+    // Variants
+    {
+      name: "Variant id",
+      mapTo: "variant.id",
+    },
+    { name: "Variant Title", mapTo: "variant.title" },
+    { name: "Variant SKU", mapTo: "variant.sku" },
+    { name: "Variant Barcode", mapTo: "variant.barcode" },
+    { name: "Variant Inventory Quantity", mapTo: "variant.inventory_quantity" },
+    { name: "Variant Allow backorder", mapTo: "variant.allow_backorder" },
+    { name: "Variant Manage inventory", mapTo: "variant.manage_inventory" },
+    { name: "Variant Weight", mapTo: "variant.weight" },
+    { name: "Variant Length", mapTo: "variant.length" },
+    { name: "Variant Width", mapTo: "variant.width" },
+    { name: "Variant Height", mapTo: "variant.height" },
+    { name: "Variant HS Code", mapTo: "variant.hs_code" },
+    { name: "Variant Origin Country", mapTo: "variant.origin_country" },
+    { name: "Variant Mid Code", mapTo: "variant.mid_code" },
+    { name: "Variant Material", mapTo: "variant.material" },
+
+    // ==== DYNAMIC FIELDS ====
+
+    // PRODUCT_OPTIONS
+    {
+      name: "Option Name",
+      match: /Option \d+ Name/,
+      // @ts-ignore
```

Comment:
todo: We don't need that if we type as the follow:
```ts
type ProductImportCsvSchema = CsvSchema<
  TParsedRowData,
  ...
>
```

Or something along that line, and therefore all the places using `Record<string, string>` for the parsed result should be using your parsed type. wdyt?

also, all the reducer should provide a return type of `TParsedRowData` to please the linter 👍 

same goes with the next columns